tests/PHPStan/Analyser/data/non-empty-string.php

```diff
@@ -134,6 +134,10 @@ public function doEmpty2(string $s): void
 		}
 	}
 
+	/** @param non-empty-string $nonEmpty */
+	public function doNegativeSubstr($nonEmpty) {
+		assertType[substr($nonEmpty, -5)]
```

Comment:
Hi, I definitely want to have `substr($nonEmpty, 0, positive-int)` covered as well. Also please investigate what happens with `substr($nonEmpty, 0, negative int)` and if it's worth covering as well.

Reply:
Per https://3v4l.org/aKs2O Negative offsets seem not to be relevant for the non-empty-string case